### PR TITLE
✨(deploy) slack notify authors of deployments

### PR DESCRIPTION
**Pull request by mlbrgl**
Send Slack notifications when a new content deployment finishes.

- [x] add SLACK_BOT_OAUTH_TOKEN to live.owid.io/live/.env
- [x] deploy https://github.com/owid/ops/pull/127
- [ ] notify developers channel new env
- [ ] add the command from the [test buildkite pipeline](https://buildkite.com/our-world-in-data/testing-notify-authors-on-slack/settings/steps) to the [deploy-code](https://buildkite.com/our-world-in-data/owid-deploy-code-master/settings/steps) and [deploy-content](https://buildkite.com/our-world-in-data/owid-deploy-code-master/settings/steps) pipelines 

To trigger a new notification locally:
- set `BUILDKITE_DEPLOY_CONTENT_PIPELINE_SLUG=testing-notify-authors-on-slack`
- set `BUILDKITE_API_ACCESS_TOKEN`
- `yarn startDeployQueueServer`
- update a chart to trigger a new "deploy" (with the queue updated above, this will only trigger a slack notification, and not an actual deploy)

This relies on:
- test buildkite pipeline: https://buildkite.com/our-world-in-data/testing-notify-authors-on-slack/settings/steps
- test lxc container, created with `owid-lxc create -t owid-cloudflare-prod matthieu-slack`, after [replacing the deploy queue `cloudflare deploy`](https://github.com/owid/ops/blob/56cd698f3e6e20d8e0d7a06534eb70def50fade1/templates/owid-cloudflare-prod/init.sh#L49) with `notify-slack` (see test pipeline)

### CODE REVIEW

**Comment by mlbrgl** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3101?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `master`
  * **PR #3100** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3100?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #3101** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3101?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/owid/owid-grapher/3101?utm_source=stack-comment).

File: ops/buildkite/deploy-content [hunk 1]
```diff
@@ -174,4 +174,4 @@ bake_gdoc_posts() {
     )
 }
 
-deploy_content
\ No newline at end of file
+deploy_content
```
Comment (mlbrgl) [inline ops/buildkite/deploy-content:177]: TODO: check this

**Comment by mlbrgl** [comment]
@Marigold thanks for the review!

> it'd be nice to have slack info in users table so that we don't have to do the conversion.

💯

> I'm just wondering why don't you send the notification directly (since you already have a connection to slack web api) and instead go through buildkite?

Yes it's true that using the slack API in two different places seems a bit redundant. I prefer going through BK because it makes the pipeline more self-contained. It also makes dealing with errors in this side-effect straightforward: we see them straightaway in context, for free (no additional reporting to set up, and no need to worry about what to report, and what it means for the rest of the code execution). 

Another reason was reusability: we might want to notify through slack from other pipelines, and it's convenient to just grap that one-liner and paste it in.

When and if we do save slack IDs in the DB, the reliance on the slack API at bake time would disappear, which might make the slacktee approach even more palatable.

> I couldn't test it on staging [because of an error](https://buildkite.com/our-world-in-data/grapher-automated-staging-environment/builds/3270#018d11a1-59a8-4688-98ed-4d2a2278aff9) (perhaps it needs to be rebased or the staging server needs to be restarted?). I

I rebuilt the server and set up the relevant `BUILDKITE_*` env vars. Upon saving or updating a new gdoc, notifications are sent.

_Note that [lightning content updates are currently broken](https://github.com/owid/owid-grapher/issues/3107), so they will appear as regular full bakes but support for lightning bakes is there:_

Screenshot 2024-01-17 at 15 42 28

**Comment by mlbrgl** [comment]
### Merge activity

* **Jan 17, 12:56 PM**: @mlbrgl started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/owid/owid-grapher/3101).

**Comment by Marigold** [comment]
Thanks for the explanation; agree with all your points.

**Pull request merged**